java/ql/src/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -27,10 +27,44 @@ class AndroidReceiveIntentMethod extends Method {
   }
 }
 
-class ContextStartActivityMethod extends Method {
-  ContextStartActivityMethod() {
-    (hasName("startActivity") or hasName("startActivities")) and
-    getDeclaringType() instanceof TypeContext
+/** A constructor of `android.content.Intent`. */
+class IntentConstructor extends TaintPreservingCallable {
+  IntentConstructor() { this.(Constructor).getDeclaringType() instanceof TypeIntent }
+
+  override predicate returnsTaintFrom(int arg) {
+    arg = 0 and this.getParameterType(arg) instanceof TypeIntent
+    or
+    arg in [1, 4] and this.getParameterType(arg) instanceof TypeClass
```

Comment:
Assuming this should match `Intent(String, URI, Context, Class)`, then this should probably be 3:
```suggestion
    arg in [1, 3] and this.getParameterType(arg) instanceof TypeClass
```

Reply:
Good catch, thanks! Fixed in fea6114.